Sources/Utilities/Security/SecurityProtectionManager/SecurityProtectionManager.swift

```diff
@@ -0,0 +1,199 @@
+    
+    private let disposeBag = DisposeBag()
+    
+}
+
+// MARK: - Public methods
+
+extension SecurityProtectionManager {
+
+    /// Checks current system status.
+    ///
+    /// In the case where device is jailbroken and there are reverse engineering
+    /// actions performed, debugger is denied and app is forced to crash.
+    func checkSystemStatus() {
+        guard !isSimulator else { return }
+        if firstLevel == nil {
+            firstLevel = SecurityService.ServiceStatus.firstLevel
+        }
+        let startupSystem = SecurityService.ServiceStatus.startSystem == .online
+        let secondLevel = SecurityService.ServiceStatus.secondLevel == .online
+        if firstLevel == .offline && (startupSystem || secondLevel) {
+            IOSSecuritySuite.denyDebugger()
+            exit(0)
+        }
+    }
+
+    
+    /// Initiates full application security which consist of multiple layers of security.
+    ///
+    /// - Parameter window: UIWindow on which blur will be applied
+    func initiateFullSecurity(in window: UIWindow?) {
+        guard !isSimulator else { return }
+        protectAppInBackground(window: window)
+        biometricChangesCheck()
+        firstLevelSystemCheck()
+        initiatePeriodicSystemCheck()
+    }
+    
+}
+
+// MARK: - Protections
+
+private extension SecurityProtectionManager {
+
+    /// Checks if device is Jailbroken when app finishes launching.
+    func firstLevelSystemCheck() {
+        UIApplication.rx.didFinishLaunching
+            .mapToVoid()
+            .subscribe(onNext: { [unowned self] in
+                checkSystemChange()
+            })
+            .disposed(by: disposeBag)
+    }
+    
+    /// Starts periodic check if device is jailbroken to detect potential reverse engineering during app runtime.
+    ///
+    /// Periodic timer is restarted every time that application becomes active.
+    func initiatePeriodicSystemCheck() {
+        UIApplication.rx.didBecomeActive
+            .startWith(())
+            .filter { [unowned self] in firstLevel == .offline }
+            .subscribe(onNext: { [unowned self] in
+                startPeriodicSystemCheck()
+            })
+            .disposed(by: disposeBag)
+    }
+
+    /// Handles background app state. Adds or removes the blur based on app state and security system states.
+    ///
+    /// - Parameter window: window on which blur will be applied
+    func protectAppInBackground(window: UIWindow?) {
+        // In modified OS version, blur everything when the app becomes inactive
+        let inModified = UIApplication.rx
+            .willResignActive
+            .filter { Self.shouldAddBlur(for: .offline) }
+            .mapTo(true)
+
+        // Non modified version is secured from the attack, so keep blur only when the app goes to the background
+        let inNonModified = UIApplication.rx
+            .didEnterBackground
+            .filter { Self.shouldAddBlur(for: .online) }
+            .mapTo(false)
```

Comment:
Is this something that we could configure? Something like `protectAppInBackground(window: UIWindow?, type: BlurProtectionLevel)` where `BlurProtectionLevel` can have something like `alwaysBlur`, `blurOnBackground`, `securityStateBase`... something like this. Then the user can decide if blur should be done on any inactivity, only when the app goes to the background or depending on the system state (as it is implemented now).